modules/javafx.controls/src/main/java/javafx/scene/control/TreeTableRow.java

```diff
@@ -439,12 +439,16 @@ private void updateItem() {
     private void updateSelection() {
         if (isEmpty()) return;
         if (index == -1 || getTreeTableView() == null) return;
-        if (getTreeTableView().getSelectionModel() == null) return;
 
-        boolean isSelected = getTreeTableView().getSelectionModel().isSelected(index);
-        if (isSelected() == isSelected) return;
+        TreeTableViewSelectionModel<T> sm = getTreeTableView().getSelectionModel();
+        if (sm == null) {
+            return;
+        }
 
-        updateSelected(isSelected);
+        boolean isSelected = !sm.isCellSelectionEnabled() && sm.isSelected(index, null);
```

Comment:
this is different from TableRow (which uses isSelected(index)) - why?

And we don't seem to have complete test coverage for row selection (otherwise this issue would have shown somewhere :) - please add test/s that fail/pass before/after the fix, respectively

Reply:
updated and added tests.